### PR TITLE
Add workflow template for cron job checking for upstream upgrades

### PR DESCRIPTION
**Pull request by guineveresaenger**
This PR adds a workflow that runs once a day on a schedule and plans to use the changes in pulumi/upgrade-provider#207 to poll upstream for upgrades, resulting in the creation of a "Upgrade terraform-provider-foo to vx.y.z" issue.

To assist during feature rollout, this PR targets 10 Tier 2 providers (incidentally these are all from a single Zapier handler that we plan to disable for careful rollout) to receive these new upgrade workflows. 

To look at a few example trials of this Workflow, [follow the links in this comment](https://github.com/pulumi/ci-mgmt/issues/724#issuecomment-1841288902)

Part of #724.

### CODE REVIEW

File: provider-ci/internal/pkg/templates/bridged-provider.config.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ fail-on-missing-mapping: true
 fail-on-extra-mapping: true
 publishRegistry: true
 checkoutSubmodules: false
+use-upstream-version-check: false # TODO: this option should be removed once all providers are onboarded to upstream version checks.
```
Comment (t0yv0) [inline provider-ci/internal/pkg/templates/bridged-provider.config.yaml:11]: Is this a new option?

File: provider-ci/internal/pkg/templates/bridged-provider/.github/workflows/check-upstream-upgrade.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    - name: "Set up git identity: name"
+      run: git config --global user.name pulumi-bot
+      shell: bash
+    - name: "Set up git identity: email"
+      run: git config --global user.email bot@pulumi.com
+      shell: bash
+    - name: Run upgrade-provider upstream check
+      id: upstream_version
+      run: |
+        upgrade-provider "$REPO" --kind=check-upstream-version
```
Comment (t0yv0) [inline provider-ci/internal/pkg/templates/bridged-provider/.github/workflows/check-upstream-upgrade.yml:33]: I see.

**Review by t0yv0 — APPROVED** [review]

**Comment by t0yv0** [comment]
🚢

File: provider-ci/internal/pkg/templates/bridged-provider.config.yaml (hunk 1 above)
Comment (guineveresaenger) [reply provider-ci/internal/pkg/templates/bridged-provider.config.yaml:11]: oof, thank you for catching that! I changed my mind and we won't be using this option, thanks to passing in a list of repos to the template [here](https://github.com/pulumi/ci-mgmt/pull/736/files#diff-aa352508a9683780662f0243439a9b38fc95a4f24455a42e87ae994f01152ce4R1)

**Pull request merged**